tls/tls_srv.c

```diff
@@ -730,22 +727,20 @@ ttls_parse_client_hello(TlsCtx *tls, unsigned char *buf, size_t len,
 	T_FSM_STATE(TTLS_CH_HS_CSLEN) {
 		BUG_ON(io->rlen >= 2);
 		if (unlikely(io->rlen)) {
-			tls->hs->tmp[TTLS_HS_TMP_STORE_SZ + 1] = *p;
-			p++;
+			tls->hs->cs_total_len += *p++;
 		}
 		else if (unlikely(buf + len - p == 1)) {
-			tls->hs->tmp[TTLS_HS_TMP_STORE_SZ] = *p;
-			p++;
+			tls->hs->cs_total_len = *p++ << 8;
```

Comment:
Is `*p++` converted to `unsigned short` before the shift? Isn't it better to convert the type explicitly?

Reply:
> Is `*p++` converted to `unsigned short` before the shift?

Well, yes and no.
Integer promotions are applied to both operands of a bit-shift operator. Since `*p` is `unsigned char`, which has lower rank than `int`, it's converted to `int`. Then result of the operation is clamped to fit into `tls->hs->cs_total_len` which is `unsigned short`. As result is always representable in `unsigned short`, no information loss occurs.
So although it's not converted to `unsigned short`, but to `int`, operation behaves as if it's `unsigned short` was here instead.

Would you like me to put explicit casts to `unsigned short` in all similar cases in this patch?

Reply:
No, I think we're good.